core/tests/protractor_utils/BlogDashboardPage.js

```diff
@@ -0,0 +1,376 @@
+    await waitFor.visibilityOf(
+      currUsername,
+      'Current user name taking too long to display');
+    await action.click('Close Preview', closeBlogCardPreviewButton);
+  };
+
+  this.waitForPublishedBlogPostsToLoad = async function() {
+    await waitFor.visibilityOf(
+      publishedBlogPostsTable, 'Blog posts table taking too long to appear.');
+  };
+
+  this.createNewBlogPost = async function() {
+    await action.click('Create Blog Post button', createBlogPostButton);
+
+    await waitFor.pageToFullyLoad();
+    await waitFor.visibilityOf(
+      blogPostTitleFieldElement,
+      'Blog Post Editor is taking too long to appear.');
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.setContent = async function(richTextInstructions) {
+    var schemaBasedEditorTag = blogPostContentEditor.element(
+      by.tagName('schema-based-editor'));
+    await waitFor.visibilityOf(
+      schemaBasedEditorTag, 'Schema based editor tag not showing up');
+    var richTextEditor = await forms.RichTextEditor(schemaBasedEditorTag);
+    await richTextEditor.clear();
+    await richTextInstructions(richTextEditor);
+    await action.click('Save Content Button', saveBlogPostContentButton);
+    await waitFor.invisibilityOf(
+      saveBlogPostContentButton,
+      'Content editor takes too long to disappear');
+  };
+
+  this.saveBlogPostAsDraft = async function(
+      blogPostTitle, richTextInstructions) {
+    await action.sendKeys(
+      'New blog post title field', blogPostTitleFieldElement, blogPostTitle);
+    await this.setContent(richTextInstructions);
+    await waitFor.visibilityOf(
+      blogPostContentDisplay, 'Blog Post content not showing up');
+    await action.click('Save as draft Button', saveBlogPostAsDraftButton);
+    await waitFor.visibilityOfSuccessToast('Blog Post Saved Successfully.');
+  };
+
+  this.publishNewBlogPost = async function(
+      blogPostTitle, richTextInstructions, tags) {
+    await action.sendKeys(
+      'New blog post title field', blogPostTitleFieldElement, blogPostTitle);
+    await this.selectTags(tags);
+    await this.setContent(richTextInstructions);
+    await waitFor.visibilityOf(
+      blogPostContentDisplay, 'Blog Post content not showing up');
+    await this.setThumbnailImage();
+    await action.click('Publish Blog Post Button', publishBlogPostButton);
+    await action.click(
+      'Confirm Publish Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast(
+      'Blog Post Saved and Published Succesfully.');
+  };
+
+  this.publishDraftBlogPost = async function(tags) {
+    await this.selectTags(tags);
+    await this.setThumbnailImage();
+    await action.click('Publish Blog Post Button', publishBlogPostButton);
+    await action.click(
+      'Confirm Publish Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast(
+      'Blog Post Saved and Published Succesfully.');
+  };
+
+  this.selectTags = async function(tags) {
```

Comment:
You had added a `waitFor.visibilityOf(blogPostTags.first(), ...)` here earlier. Why was it removed? I thought it was a good thing to have

Reply:
Opps, Sorry. I don't remember how did it get removed. Sorry for negligence